pkg/applicationserver/config.go

```diff
@@ -309,3 +310,14 @@ func (c EndDeviceLocationStorageConfig) NewRegistry(ctx context.Context, comp *c
 	}
 	return registry, nil
 }
+
+// LastSeenConfig defines configuration for the device last seen map which stores timestamps for batch updates.
+type LastSeenConfig struct {
+	MapCapacity int           `name:"map-capacity" description:"Capacity of last seen map"`
+	Interval    time.Duration `name:"flush-interval" description:"Interval at which lat seen timestamps are sent to Identity Server"`
+}
+
+// NewLastSeen defines a new batch update map.
+func (c LastSeenConfig) NewLastSeen(ctx context.Context, comp *component.Component) (*LastSeenProvider, error) {
+	return NewLastSeen(ctx, c.MapCapacity, time.NewTicker(c.Interval).C, comp)
```

Comment:
Validate `MaxCapacity`/`Interval` here. `NewTicker(0)` panics for example.

Having both equal to `0` could return a no-op implementation (and you can use an interface).